cmd/obslytics/export_e2e_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/cortexproject/cortex/integration/e2e"
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+	"github.com/thanos-community/obslytics/pkg/input"
+	"github.com/thanos-community/obslytics/pkg/input/storeapi"
+	"github.com/thanos-community/obslytics/pkg/output"
+	"github.com/thanos-community/obslytics/pkg/output/parquet"
+	"github.com/thanos-io/thanos/pkg/http"
+
+	"github.com/thanos-io/thanos/pkg/testutil"
+	"github.com/thanos-io/thanos/test/e2e/e2ethanos"
+)
+
+// defaultPromConfig returns Prometheus config that sets Prometheus to:
+// * expose 2 external labels, source and replica.
+// * scrape fake target. This will produce up == 0 metric which we can assert on.
+// * optionally remote write endpoint to write into.
+func defaultPromConfig(name string, replica int, remoteWriteEndpoint, ruleFile string) string {
+	config := fmt.Sprintf(`
+global:
+  external_labels:
+    prometheus: %v
+    replica: %v
+scrape_configs:
+- job_name: 'myself'
+  # Quick scrapes for test purposes.
+  scrape_interval: 1s
+  scrape_timeout: 1s
+  static_configs:
+  - targets: ['localhost:9090']
+`, name, replica)
+
+	if remoteWriteEndpoint != "" {
+		config = fmt.Sprintf(`
+%s
+remote_write:
+- url: "%s"
+  # Don't spam receiver on mistake.
+  queue_config:
+    min_backoff: 2s
+    max_backoff: 10s
+`, config, remoteWriteEndpoint)
+	}
+
+	if ruleFile != "" {
+		config = fmt.Sprintf(`
+%s
+rule_files:
+-  "%s"
+`, config, ruleFile)
+	}
+
+	return config
+}
+
+func TestThanos_Parquet_e2e(t *testing.T) {
+	t.Parallel()
+
+	s, err := e2e.NewScenario("e2e_test_thanos_parquet")
+	testutil.Ok(t, err)
+	t.Cleanup(e2ethanos.CleanScenario(t, s))
+
+	now := time.Now()
+
+	// TODO(bwplotka): Allow clients to specify image directly via function args.
+	testutil.Ok(t, os.Setenv("THANOS_IMAGE", "quay.io/thanos/thanos:v0.15.0"))
+	prom, sidecar, err := e2ethanos.NewPrometheusWithSidecar(s.SharedDir(), s.NetworkName(), "1", defaultPromConfig("test", 0, "", ""), e2ethanos.DefaultPrometheusImage())
+	testutil.Ok(t, err)
+	testutil.Ok(t, s.StartAndWaitReady(prom, sidecar))
+
+	testutil.Ok(t, prom.WaitSumMetricsWithOptions(e2e.Greater(512), []string{"prometheus_tsdb_head_samples_appended_total"}, e2e.WaitMissingMetrics))
+
+	logger := log.NewLogfmtLogger(os.Stderr)
+	t.Run("export up metric to parquet file", func(t *testing.T) {
```

Comment:
@bwplotka Can you add `using StoreAPI` in this run string? Just to make it clear which input method we are using

Reply:
Makes sense to add that once we have other inputs (: